### PR TITLE
vconsole: permit configuration of vconsole settings via credentials

### PR DESCRIPTION
**Pull request by poettering**
Inspired by: https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/5T5KYCTLGSCX4KNEYCIPYYZUJZK5P6F3/

Note the relationship to the pre-existing firstboot.* creds. 

The firstboot creds are propagated into the system on first boot, after the initrd transition These new creds act as fallback defaults even before the initrd transition.

### CODE REVIEW

**Comment by poettering** [comment]
(needs docs still)

**Comment by poettering** [comment]
docs added

**Comment by bluca** [comment]
```
stderr:
/* test_read_credential_strings */

=================================================================
==24712==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x556355fa8666 in __interceptor_realloc (/home/runner/work/systemd/systemd/build/test-creds+0xa3666) (BuildId: ec611504869f01a3cf900822b054c24c703b18ca)
    #1 0x7f3e5aa3d9b2 in path_extend_internal /home/runner/work/systemd/systemd/build/../src/basic/path-util.c:558:14
    #2 0x556355fe33e8 in test_read_credential_strings /home/runner/work/systemd/systemd/build/../src/test/test-creds.c:45:9
    #3 0x556355fe4947 in run_test_table /home/runner/work/systemd/systemd/build/../src/shared/tests.h:106:33
    #4 0x556355fe4947 in main /home/runner/work/systemd/systemd/build/../src/test/test-creds.c:73:1
    #5 0x7f3e59f1fd8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f) (BuildId: 69389d485a9793dbe873f0ea2c93e02efaa9aa3d)

SUMMARY: AddressSanitizer: 32 byte(s) leaked in 1 allocation(s).
```

File: man/systemd-firstboot.xml [hunk 1]
```diff
@@ -325,7 +325,15 @@
         <term><literal>firstboot.keymap</literal></term>
 
         <listitem><para>This credential specifies the keyboard setting to set during first boot, in place of
-        prompting the user.</para></listitem>
+        prompting the user.</para>
+
+        <para>Note the relationship to the <literal>vconsole.keymap</literal> credential understood by
+        <citerefentry><refentrytitle>systemd-vconsole-setup.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>:
+        both ultimately effect the same setting, but <varname>firstboot.keymap</varname> is written into
```
Comment (bluca) [inline man/systemd-firstboot.xml]: /effect/affect/

File: man/systemd-vconsole-setup.service.xml [hunk 2]
```diff
@@ -49,6 +49,44 @@
...
+        <term><varname>vconsole.keymap</varname></term>
+        <term><varname>vconsole.keymap_toggle</varname></term>
+
+        <listitem><para>The keymap (and toggle keymap) to apply. The matching options in
+        <filename>vconsole.conf</filename> and on the kernel command line take precedence over these
+        credentials.</para>
+
+        <para>Note the relationship to the <varname>firstboot.keymap</varname> credential understood by
+        <citerefentry><refentrytitle>systemd-firstboot.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>:
+        both ultimately effect the same setting, but <varname>firstboot.keymap</varname> is written into
```
Comment (bluca) [inline man/systemd-vconsole-setup.service.xml]: /effect/affect/

**Comment by poettering** [comment]
fixed, ptal

**Review by bluca — APPROVED** [review]

**Pull request merged**